src/StoreKeeper/WooCommerce/B2C/Imports/ProductImport.php

```diff
@@ -1039,11 +1039,74 @@ function ($attribute) {
                     return !key_exists('attribute_option_id', $attribute);
                 }
             );
-            foreach ($nonAttributeOptions as $attribute) {
+            // Find attributes that have 'attribute_option_color_hex'
+            $colorAttributes = array_filter(
+                $dotObject->get('flat_product.content_vars'),
+                function ($attribute) {
+                    return array_key_exists('attribute_option_color_hex', $attribute);
+                }
+            );
+
+            $mergedAttributes = array_merge($nonAttributeOptions, $colorAttributes);
+
+            $blocksyTaxonomyMetaOptions = [];
+
+            foreach ($mergedAttributes as $attribute) {
                 if (key_exists('attribute_id', $attribute)) {
+                    $BlogModule = $this->storekeeper_api->getModule('BlogModule');
+                    $response = $BlogModule->listTranslatedAttributes(
+                        0,
+                        0,
+                        1,
+                        [
+                            [
+                                'name' => 'id',
+                                'dir' => 'desc',
+                            ],
+                        ],
+                        [
+                            [
+                                'name' => 'id__=',
+                                'val' => $attribute['attribute_id'],
+                            ],
+                        ]
+                    );
+
+                    $data = $response['data'][0];
+                    $attributeType = $data['type'];
```

Comment:
wrap inside a function `getAttributeType(): string`, use lazy loading to make the call only once per attribute id